openmeetings-core/src/main/java/org/apache/openmeetings/core/remote/KurentoHandler.java

```diff
@@ -120,7 +121,8 @@ public void init() {
 			try {
 				kuid = randomUUID().toString();
 				client = KurentoClient.create(kurentoWsUrl, new KConnectionListener(kuid));
-				client.getServerManager().addObjectCreatedListener(new KWatchDog());
+				client.getServerManager().addObjectCreatedListener(new KWatchDogCreate());
+				client.getServerManager().addObjectDestroyedListener(new KWatchDogDestroy());
```

Comment:
I think the destroy events are very useful in terms of detecting that the stream actually closed. So from a debug perspective this is very useful.
And I think we can also continue to use the destroy event to detect a KStream is gone. Rather then just rely on browser events.

Reply:
We are not relying on browser events already
https://github.com/apache/openmeetings/blob/master/openmeetings-core/src/main/java/org/apache/openmeetings/core/remote/KStream.java#L123

The stream should both be stopped and destroyed
I would further investigate why it is not happening

Reply:
Well when stopping to recording tis won't trigger. And when, after stopping to record, leave the room or exit the browser, it won't seem to trigger either.

Reply:
Recording is `virtual` event, it is only flag on current stream
we need to be able to set/unset this flag any time, and the stream will not be affected

Reply:
you are demanding 😄

Reply:
let me search further

Reply:
It's not me :(
here is the example: https://issues.apache.org/jira/browse/OPENMEETINGS-2253